File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Adapter/Cpp/src/USBCToLightningAdapter.cpp
#include <iostream>
#include "../include/USBCToLightningAdapter.h"

USBCToLightningAdapter::USBCToLightningAdapter(USBCCharger* charger) : usbcCharger(charger) {}

void USBCToLightningAdapter::connectWithLightning() {
    std::cout << "Using USB-C to Lightning adapter..." << std::endl;
    usbcCharger->connectWithUSBC();
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Adapter/Java/USBCToLightningAdapter.java
public class USBCToLightningAdapter implements LightningPort {
    private USBCCharger usbcCharger;

    public USBCToLightningAdapter(USBCCharger usbcCharger) {
        this.usbcCharger = usbcCharger;
    }

    @Override
    public void connectWithLightning() {
        System.out.println("Using USB-C to Lightning adapter...");
        usbcCharger.connectWithUSBC();
    }
}
